### PR TITLE
Add post about stabilizing intra-doc links

### PR DESCRIPTION
**Pull request by jyn514**
cc @Manishearth 
I set the date as tomorrow, but I don't have strong opinions about when to publish.

### CODE REVIEW

**Comment by Manishearth** [comment]
I'm hoping to hold off until we know when intra doc is going to stabilize

**Comment by Manishearth** [comment]
cc @QuietMisdreavus if you wanna take a look

File: posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md
Comment (GuillaumeGomez) [inline posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md:9]: Isn't punctuation missing here?

File: posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md
Comment (GuillaumeGomez) [inline posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md:68]: Maybe put this error into another codeblock so the highlighting isn't weird?

File: posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md
Comment (Manishearth) [reply posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md:9]: oh it seems like some of the text got dropped @jyn514

Comment (jyn514) [reply posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md:9]: Hmm, that's what I see on the HackMD:

image

Did you mean to write more?

Comment (Manishearth) [reply posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md:9]: Alright, I must have messed up. This should be something like "We're excited to share that intra-doc links are stabilizing soon!"

**Comment by Manishearth** [comment]
Should we mention somewhere that the feature has been auto enabled on nightly and thus on docs.rs somewhere? (And that this blog post is a way to signal "you really should start using this now")

**Comment by GuillaumeGomez** [comment]
@Manishearth Oh that's a good point. I think we should definitely mention it.

**Comment by jyn514** [comment]
> Should we mention somewhere that the feature has been auto enabled on nightly and thus on docs.rs somewhere? (And that this blog post is a way to signal "you really should start using this now")

I'm not really comfortable encouraging people to use unstable features ... In general it makes me uncomfortable that all docs are built with nightly by default, but I'm not sure how docs.rs can change that without breaking the entire ecosystem of docs.

**Comment by Manishearth** [comment]
@jyn514 no, you misunderstand, we're encouraging them to use it more now that it is stable.

**Comment by jyn514** [comment]
Oh I see, we're saying to skip the 12-week delay until it hits stable 😆 I guess that works, I'll add that.

**Comment by Manishearth** [comment]
@jyn514 no no!

The reason I want that in is that some people will be confused as to why we're talking about a feature that has been around forever. This clarifies what has changed, and the important thing is that now we strongly recommend people use it. It's not about the 12 week period, in fact we should probably publish this after the feature hits beta.

**Comment by jyn514** [comment]
I pushed a change but I'm not super confident in the wording ... maybe you should write this bit :P

File: posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md
Comment (Manishearth) [inline posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md:19]:
```suggestion
Intra-doc links have been around for a while, all the way back [since 2017][tracking-issue]! They have been available on `nightly` without flags (and thus, on docs.rs), so you may be surprised to hear that they weren't yet stable. What's changing now is that they will be available on stable Rust, which also means we are more confident in the implementation and would strongly encourage their use. We recommend that you switch your libraries to use intra-doc links, which will fix broken links for re-exported types and links to different crates. We hope to add support for automating this process with [`cargo fix`] in the future.
```

File: posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md
Comment (camelid) [inline posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md:7]: I think this is handled by the front matter, no?

```suggestion
```

File: posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md
Comment (camelid) [inline posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md:11]:
```suggestion
[Intra-doc links] are a feature of `rustdoc` that allow you to link to '[items]' - functions, types, and more - by their name, instead of a hard-coded URL. This lets you have accurate links even if your types are [re-exported in a different module or crate][broken-string-links]. Here is a simple example:
```

File: posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md
Comment (camelid) [inline posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md:75]:
```suggestion
(If you're not interested in the internals of the Rust compiler, feel free to skip this section.)
```

File: posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md
Comment (camelid) [inline posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md:78]:
```suggestion
The `everybody_loops` pass turns the above code into:
```

**Comment by camelid** [comment]
Might it be better to post this on the main Rust blog so that more people see it?

File: posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md
Comment (camelid) [reply posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md:7]: image

File: posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md
Comment (camelid) [inline posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md:16]: I would suggest adding a blank line for more visual clarity.

```suggestion
/// Link to [`f()`]
pub struct S;

pub fn f() {}
```

File: posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md
Comment (camelid) [inline posts/inside-rust/2020-09-11-stabilizing-intra-doc-links.md:19]: Link to docs.rs?

```suggestion
Intra-doc links have been around for a while, all the way back [since 2017][tracking-issue]! They have been available on `nightly` without flags (and thus, on [docs.rs](https://docs.rs)), so you may be surprised to hear that they weren't yet stable. What's changing now is that they will be available on stable Rust, which also means we are more confident in the implementation and would strongly encourage their use. We recommend that you switch your libraries to use intra-doc links, which will fix broken links for re-exported types and links to different crates. We hope to add support for automating this process with [`cargo fix`] in the future.
```

**Comment by jyn514** [comment]
Not sure after last time ... @pietroalbini what do you think about posting this on the main blog instead of inside-rust?

**Comment by Manishearth** [comment]
I'm okay with either, personally. Inside rust doesn't really require approval, but for the main blog the core team should consider this important enough. Idk if they will.

**Comment by pietroalbini** [comment]
This seems more like a "behind the scenes" post on the history of the feature and the challenges implementing it, rather than an announcement of what the feature is and when it will be stabilized.

I *personally* think that's content better suited for Inside Rust, but I'm happy to raise the question to the rest of the Core Team if you so wish.

**Comment by jyn514** [comment]
I agree with that, let's keep it on inside-rust.

**Comment by Manishearth** [comment]
Seems good. The only reason it might work on the main blog is because it is an _interesting_ journey, but that's not necessarily the right bar.

**Comment by jyn514** [comment]
This week in rust just posted about the stabilization PR, so I think we should publish this today.

**Review by pietroalbini — APPROVED** [review]

**Pull request merged**
